cli/azd/cmd/account.go

```diff
@@ -0,0 +1,229 @@
+					},
+					{
+						Heading:       "Regional Name",
+						ValueTemplate: "{{.RegionalDisplayName}}",
+					},
+				},
+			}
+
+			err = formatter.Format(locations, writer, tableOptions)
+		}
+
+		if err != nil {
+			return fmt.Errorf("failed formatting output")
+		}
+
+		return nil
+	})
+
+	return commands.Build(action, rootOptions, "list-locations", "Gets the available Azure locations for the default Azure account.", nil)
+}
+
+// Command that shows the default subscription & location for the logged in user
+func accountShowCmd(rootOptions *internal.GlobalCommandOptions) *cobra.Command {
+	action := commands.ActionFunc(func(ctx context.Context, cmd *cobra.Command, args []string, azdCtx *azdcontext.AzdContext) error {
+		formatter := output.GetFormatter(ctx)
+		writer := output.GetDefaultWriter()
+		manager := account.NewManager(ctx)
+
+		config, err := manager.GetAccountDefaults(ctx)
+		if err != nil {
+			return fmt.Errorf("failed retrieving account defaults: %w", err)
+		}
+
+		err = formatter.Format(config, writer, nil)
+		if err != nil {
+			return fmt.Errorf("failed formatting output: %w", err)
+		}
+
+		return nil
+	})
+
+	return commands.Build(action, rootOptions, "show", "Shows the default Azure subscription", nil)
+}
+
+func accountClearCmd(rootOptions *internal.GlobalCommandOptions) *cobra.Command {
+	action := commands.ActionFunc(func(ctx context.Context, cmd *cobra.Command, args []string, azdCtx *azdcontext.AzdContext) error {
+		console := input.GetConsole(ctx)
+		manager := account.NewManager(ctx)
+
+		err := manager.Clear(ctx)
+		if err != nil {
+			return errors.New("failed clearing AZD account defaults")
+		}
+
+		console.Message(ctx, output.WithSuccessFormat("SUCCESS: Azure Developer CLI defaults have been reset."))
+
+		return nil
+	})
+
+	return commands.Build(action, rootOptions, "clear", "Clears all defaults from Azure Developer CLI configuration.", nil)
+}
+
+// Options for account set command
+type accountSetAction struct {
+	rootOptions    *internal.GlobalCommandOptions
+	subscriptionId string
+	location       string
+}
+
+func accountSetCmd(rootOptions *internal.GlobalCommandOptions) *cobra.Command {
+	return commands.Build(
+		&accountSetAction{
+			rootOptions: rootOptions,
+		}, rootOptions,
+		"set",
+		"Sets the default Azure subscription and/or location.",
+		nil)
+}
+
+func (a *accountSetAction) SetupFlags(persis, local *pflag.FlagSet) {
+	local.StringVarP(&a.subscriptionId, "subscriptionId", "s", "", "Azure Subscription ID.")
+	local.StringVarP(&a.location, "location", "l", "", "Azure location.")
+}
+
+func (a *accountSetAction) Run(ctx context.Context, cmd *cobra.Command, args []string, azdCtx *azdcontext.AzdContext) error {
+	console := input.GetConsole(ctx)
+	manager := account.NewManager(ctx)
+
+	// Sets defaults subscription when -s / --subscriptionId argument has been specified
+	if strings.TrimSpace(a.subscriptionId) != "" {
+		_, err := manager.SetDefaultSubscription(ctx, a.subscriptionId)
```

Comment:
I think this needs to accept a sub name too.

Reply:
Good call - i will add this.